src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -1282,6 +1309,62 @@ bool CommandLineInterface::GenerateOutput(
   return true;
 }
 
+bool CommandLineInterface::GenerateDependencyManifestFile(
+    const vector<const FileDescriptor*> parsed_files,
+    DiskSourceTree * source_tree) {
+  FileDescriptorSet file_set;
+
+  set<const FileDescriptor*> already_seen;
+  for (int i = 0; i < parsed_files.size(); i++) {
+    GetTransitiveDependencies(parsed_files[i],
+                              false,
+                              &already_seen, file_set.mutable_file());
+  }
+
+  int fd;
+  do {
+    fd = open(manifest_name_.c_str(),
+              O_WRONLY | O_CREAT | O_TRUNC | O_BINARY, 0666);
+  } while (fd < 0 && errno == EINTR);
+
+  if (fd < 0) {
+    perror(manifest_name_.c_str());
+    return false;
+  }
+
+  stringstream ss;
+  string output_filename = manifest_name_;
+  if (output_filename.compare(0, 2, "./") == 0) {
+    output_filename = output_filename.substr(2);
+  }
+  ss << output_filename << ": ";
+  for (set<const FileDescriptor*>::const_iterator it = already_seen.begin(); it != already_seen.end(); ++it ) {
```

Comment:
Yes this should be file_set.  Originally I was calling GetSourceLocation, which is a private member of FileDescriptor.